man/tamper.Rd

```diff
@@ -0,0 +1,59 @@
+% Generated by roxygen2 (4.1.1): do not edit by hand
+% Please edit documentation in R/tamper.R
+\name{tamper}
+\alias{tamper}
+\title{Investigate a pipe, after an error}
+\usage{
+tamper()
+}
+\description{
+\code{tamper} is a function that can be used as an error callback,
+similarly to \code{utils::recover}. Tamper is pipe-friendly: it will
+show you the exact place of the error in the pipe.
+}
+\section{Example}{
+
+
+After running the following code: \preformatted{options(error = tamper)
+
+f <- function(data) {
+   data \%>\%
+     (function(x) force(x)) \%>\%
+     multiply_by(10) \%>\%
+     add(10) \%>\%
+     add("oh no!") \%>\%
+     subtract(5) \%>\%
+     divide_by(5)
+}
+
+ 1:10 \%>\%
+   multiply_by(2) \%>\%
+   f() \%>\%
+   add(1:10)
+}
+you will see: \preformatted{Error in add(., "oh no!") : non-numeric argument to binary operator
+
+Enter a pipe stage number, 1 to switch mode, or 0 to exit
```

Comment:
If I enter a pipe stage number here (e.g., 6), the browser is in the `freduce` function. Is there a reason why the browser is not one level deeper, in the function `function_list[[1L]]`? Also, is there a point in exploring the frames that are below the actual error?